### DISCUSSION TITLE
Search Users Route

### DISCUSSION THREAD

**Discussion by fubuloubu** [fief-dev/fief#395] [Ideas]
I want to add a capability to allow searching through users, so that I can find another user to reference for client logic (access control)

It would be helpful to be able to search by a number of different fields, but some fields should require an exact match (such as an email, you don't want to leak a user account unless their full email is known), where other fields it may be better to allow a subset match (such as the first couple of characters of a username). Some fields it might not make sense to allow search at all, or restrict search for privacy reasons.

Also feel like this feature should be configurable, so you can choose to allow it if your application(s) need it.

This may be out of scope of what Fief is meant to do, I'm not sure if it's just better to directly query the database for this (but doing that would add additional codependency that doesn't seem great)

**Comment by fubuloubu** [comment]
Discovered a decent workaround: implement indexes in your [caching method](https://docs.fief.dev/integrate/python/fastapi/#caching-user-information) for the field(s) you'd like to search over

It may be slightly out of date at times, but that's probably good enough for most applications.

Still think it would be good to have user search direct from the source, but this gets me close enough for now

**Comment by frankie567** [comment]
Interesting! I think we could imagine having an API allowing query parameters like this: `/users/?fields_name=abc&fields_accepted_terms=true`. I added it to the backlog: https://github.com/orgs/fief-dev/projects/1/views/4?pane=issue&itemId=70859816

That said, if the field you want to look for is critical to provide your business logic, I would recommend to keep it on your side in your own DB.

**Reply by fubuloubu** [reply]
As far as workflows, my search is basically being used to add people to a dowmstream resource by ID. It would be good to have both "fuzzy" partial match and complete match for this e.g. partial match for fields like a username, or a complete match against a user's email (to prevent harvesting user's emails)
